### ISSUE TITLE
Using rx-state as global state (best practice)

### ISSUE THREAD

**Issue by andreElrico** [BioPhoton/rx-angular#111]
What is the best practice to use rxState as global state?

I have this setup currently.

+ create injection token 

`export const GLOBAL_RX_STATE = new InjectionToken('GLOBAL_RX_STATE');`

--------------------------------------------------
`in app.module.ts`

create instance with initial state

```
const _RX_STATE_INSTANCE =
(function(initialValue) {
    const ans = new RxState();
    ans.set(initialValue);

    return ans;
})(INITIAL_STATE);
```

+ add it to the providers in `app.module`'s decorator

```
    providers: [
        ...
        { provide: GLOBAL_RX_STATE, useValue: _RX_STATE_INSTANCE},
        ...
    ],
```

**Comment by Karnaukhov-kh** [comment]
Hi.

Just a suggestion. You can have a service with ({providedIn: 'root'}) that extends RxState.

Example: https://github.com/BioPhoton/rx-angular/blob/7d9d6999f491915f9ab39f832c717ca70e0c2c47/libs/ngx-rx-state/src/lib/rx-global-state.service.ts

**Comment by BioPhoton** [comment]
related to #113

**Comment by nartc** [comment]
![image](https://user-images.githubusercontent.com/25516557/83892325-2169fc80-a714-11ea-8654-39d9c497dadc.png)
Have your `providedIn: 'root'` service extends RxState. Works super nicely.

**Comment by BioPhoton** [comment]
closed in favor of #113
